src/payment/strategies/braintree/braintree-credit-card-payment-strategy.ts

```diff
@@ -46,34 +54,46 @@ export default class BraintreeCreditCardPaymentStrategy implements PaymentStrate
         return this._store.getState();
     }
 
-    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const { payment, ...order } = orderRequest;
+    async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const state = await this._store.dispatch(
+            this._orderActionCreator.submitOrder(omit(orderRequest, 'payment'), options)
+        );
+
+        const {
+            billingAddress: { getBillingAddressOrThrow },
+            order: { getOrderOrThrow },
+            payment: { isPaymentDataRequired },
+        } = state;
 
-        if (!payment) {
-            throw new PaymentArgumentInvalidError(['payment']);
+        if (!isPaymentDataRequired(orderRequest.useStoreCredit) || !orderRequest.payment) {
```

Comment:
if `!orderRequest.payment`shouldn't we throw `PaymentArgumentInvalidError` ? and not create the order?

Reply:
Yes, I accidentally removed the guard 😄

Reply:
I don't think you need `|| !orderRequest.payment` in here?

Also, any reason why you removed the destructuring (`const { payment, ...order } = orderRequest;`)? you might not need to use `omit`

Reply:
Good spot. Regarding the destructuring, no particular reason - I think at one point I moved the code to a different place and then I changed my mind but didn't revert cleanly.